### ISSUE TITLE
Refactoring code to replace Keras utilities

### ISSUE THREAD

**Issue by Benjamintdk** [orcasound/orcaal#33]
# Issue
Currently, the code in the `train_and_predict` service relies heavily on built-in Keras API such as [`ImageDataGenerator`](https://github.com/Benjamintdk/orcaal/blob/0f2f6fc4b5c4227db5ec48aca372301974ff5a7d/train_and_predict/train.py#L45)  and [`flow_from_directory`](https://github.com/Benjamintdk/orcaal/blob/0f2f6fc4b5c4227db5ec48aca372301974ff5a7d/train_and_predict/train.py#L52) function. While this is acceptable for online tutorials and as a teaching tool, it is not so suitable for a production ready environment. This is because it is bad practice for the code to be interacting directly with the data, and having direct write/update access to the data. Also, the `flow_from_directory` function requires the data to be segregated into folders, which complicates matters and makes it difficult for future 3rd party users, who will have to adapt their data format accordingly. The solution to this problem is multi-facted, and should be discussed before any implementation is done.

# Possible solution
- One possible fix would be to update the Postgres database schema (or add a new one) which keeps the filenames of the data, and their respective labels. 
- A sampling technique can then be used to split the labeled files into train and validation sets for retraining when sufficient new examples have been labelled.

# Possible tasks to do
- Design new setup replacing the `flow_from_directory` function and `ImageDataGenerator`
- Update/Create a new schema to store data state in the Postgres database instead.
